source/opt/instruction.h

```diff
@@ -742,6 +750,11 @@ inline void Instruction::SetResultType(uint32_t ty_id) {
   operands_.front().words = {ty_id};
 }
 
+inline void Instruction::ForceResultType(uint32_t ty_id) {
+  SetResultType(ty_id);
+  operands_.front().type = SPV_OPERAND_TYPE_TYPE_ID;
+}
+
```

Comment:
This seems like you will be overriding the operand that was in there. Since it is only used on a single spot, can't you do it differently? I'll comment more on the place it is used.